src/tests/ftest/control/dmg_system_reformat.py

```diff
@@ -0,0 +1,129 @@
+from general_utils import human_to_bytes
+
+
+class DmgSystemReformatTest(TestWithServers):
+    # pylint: disable=too-many-ancestors
+    """Test Class Description:
+
+    Test to verify dmg storage format reformat option works as expected on a
+    DAOS system after a controlled shutdown.
+
+    :avocado: recursive
+    """
+
+    def setUp(self):
+        """Set up for DmgSystemReformatTest test."""
+        super(DmgSystemReformatTest, self).setUp()
+
+        host = self.hostlist_servers[0]
+        scm_list = self.server_managers[-1].get_config_value("scm_list")
+        pmem_ns = os.path.split(scm_list[-1])
+        storage_info = self.get_dmg_command().storage_scan(verbose=True)
+        self.scm_cap = storage_info[host]["scm"][pmem_ns[-1]]["capacity"]
+
+    def create_pool_at_capacity(self, percentage):
+        """Create a pool at specified capacity.
+
+        Args:
+            percentage (int): percent to use of total space available for pool.
+                i.e. 0.5 (50%) 0.25(25%)
+
+        Raises:
+            CommandFailure: raise exception if pool creation fails.
+
+        Returns:
+            TestPool: the created test pool object.
+
+        """
+        if percentage > 1 or percentage <= 0:
+            self.fail("The percent value provided cannot be used: %s percent",
+                      percentage * 100)
+        # Convert info from dmg's human readable to bytes
+        scm_size = int(percentage * human_to_bytes(self.scm_cap))
+
+        # Update size value with unit and create, create pool object.
+        self.log.info("Create pool at %s percent capacity.", percentage * 100)
+        pool = self.get_pool(create=False, connect=False)
+        pool.scm_size.update(scm_size)
+        pool.create()
+
+        return pool
+
+    @fail_on(ServerFailed)
+    def test_dmg_system_reformat(self):
+        """
+        JIRA ID: DAOS-5415
+
+        Test Description: Test dmg system reformat functionality.
+
+        :avocado: tags=all,small,pr,hw,control,sys_reformat,dmg
+        """
+        self.create_pool_at_capacity(0.7)
+
+        self.log.info("Check that new pool will fail with DER_NOSPACE")
+        try:
+            pool = self.create_pool_at_capacity(0.7)
+        except TestFail as err:
+            self.log.info("Pool create failed: %s", str(err))
+            if pool.dmg.result.stderr:
+                if "DER_NOSPACE" not in pool.dmg.result.stderr:
+                    self.fail("Pool create did not fail do to DER_NOSPACE!")
+
+        self.log.info("Stop running io_server instancess: 'dmg system stop'")
+        self.get_dmg_command().system_stop(force=True)
+        if self.get_dmg_command().result.exit_status != 0:
+            self.fail("Detected issues performing a system stop: {}".format(
+                self.get_dmg_command().result.stderr))
+
+        self.log.info("Perform dmg storage format on all system ranks:")
+        self.get_dmg_command().storage_format(reformat=True)
```

Comment:
Is there a way we can set the DMG config file hostlist to empty before running this command? That will ensure the membership information is being used to initiate the reformat rather than explicit hostlist in the DMG config file. Seems like the right way to verify system reformat behaviour without invasive monitoring which might make the tests brittle.

Reply:
Yes, I'll make a fix with this suggestion.